### PR TITLE
CDAP-11470 Add jul-to-slf4j bridge

### PR DESCRIPTION
**Pull request by deepakwadhwani**

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java [hunk 1]
```diff
@@ -994,6 +995,7 @@ public boolean accept(String className, URL classUrl, URL classPathUrl) {
     classes.add(MapReduce.class);
     classes.add(MapperWrapper.class);
     classes.add(ReducerWrapper.class);
+    classes.add(org.slf4j.bridge.SLF4JBridgeHandler.class);
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java]: Why full name?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/distributed/MapReduceContainerLauncher.java [hunk 2]
```diff
@@ -61,6 +62,9 @@ public static void launch(String mainClassName, String[] args) {
       throw new IllegalStateException("Failed to remove main class resource " + resource);
     }
 
+    // Install the JUL to SLF4J Bridge
+    SLF4JBridgeHandler.install();
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/distributed/MapReduceContainerLauncher.java]: It is better to load this class from the MainClassLoader constructed below and use reflection to invoke the method

**Review by chtyim — COMMENTED** [review]
You should similar changes to the SparkContainerLauncher

**Comment by deepakwadhwani** [comment]
@chtyim thank you for your comments, please take another look.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/distributed/MapReduceContainerLauncher.java [hunk 3]
```diff
@@ -65,6 +65,15 @@ public static void launch(String mainClassName, String[] args) {
     URL[] classLoaderUrls = urls.toArray(new URL[urls.size()]);
     ClassLoader mainClassLoader = new MainClassLoader(classLoaderUrls, systemClassLoader.getParent());
 
+    // Install the JUL to SLF4J Bridge
+    try {
+      mainClassLoader.loadClass("org.slf4j.bridge.SLF4JBridgeHandler")
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/distributed/MapReduceContainerLauncher.java]: It's ok to use `SLF4JBridgeHandler.class.getName()` here.

**Comment by chtyim** [comment]
Are you going to add the call in `SparkContainerLauncher` as well?

**Comment by deepakwadhwani** [comment]
@chtyim Added bridge to spark launcher and moved logger "com.sun.jersey"to "WARN". Please take another look.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/distributed/MapReduceContainerLauncher.java [hunk 4]
```diff
@@ -65,6 +66,15 @@ public static void launch(String mainClassName, String[] args) {
     URL[] classLoaderUrls = urls.toArray(new URL[urls.size()]);
     ClassLoader mainClassLoader = new MainClassLoader(classLoaderUrls, systemClassLoader.getParent());
 
+    // Install the JUL to SLF4J Bridge
+    try {
+      mainClassLoader.loadClass(SLF4JBridgeHandler.class.getName())
+        .getDeclaredMethod("install")
+        .invoke(null);
+    } catch (Exception e) {
+      throw new RuntimeException("Failed to invoke SLF4JBridgeHandler.install() required for jul-to-slf4j bridge", e);
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/distributed/MapReduceContainerLauncher.java]: A failure in setting logger shouldn't cause failure of the job.

File: cdap-master/pom.xml [hunk 5]
```diff
@@ -201,6 +201,10 @@
       <type>test-jar</type>
       <scope>test</scope>
     </dependency>
+    <dependency>
```
Comment (chtyim) [inline cdap-master/pom.xml]: Non-test dependency should be before test dependency.

File: cdap-spark-core-base/src/main/java/co/cask/cdap/app/runtime/spark/distributed/SparkContainerLauncher.java [hunk 6]
```diff
@@ -66,6 +67,15 @@ public static void launch(String mainClassName, String[] args) {
     URL[] classLoaderUrls = urls.toArray(new URL[urls.size()]);
     ClassLoader classLoader = new SparkContainerClassLoader(classLoaderUrls, systemClassLoader.getParent());
 
+    // Install the JUL to SLF4J Bridge
+    try {
+      classLoader.loadClass(SLF4JBridgeHandler.class.getName())
+        .getDeclaredMethod("install")
+        .invoke(null);
+    } catch (Exception e) {
+      throw new RuntimeException("Failed to invoke SLF4JBridgeHandler.install() required for jul-to-slf4j bridge", e);
```
Comment (chtyim) [inline cdap-spark-core-base/src/main/java/co/cask/cdap/app/runtime/spark/distributed/SparkContainerLauncher.java]: Same as above.

**Comment by chtyim** [comment]
LGTM

**Comment by deepakwadhwani** [comment]
Clean build: https://builds.cask.co/browse/CDAP-DUT5661-14

**Pull request merged**